wantedattributebuilder.go

```diff
@@ -65,6 +65,10 @@ func (builder *WantedAttributeBuilder) Build() WantedAttribute {
 
 // MarshalJSON returns the JSON encoding
 func (attr *WantedAttribute) MarshalJSON() ([]byte, error) {
+	err := notEmpty(attr.name, "Wanted attribute names must not be empty")
```

Comment:
We can put this validation in the `.Build()` function can't we, to catch it earlier?

Reply:
The Build function doesn't return an error - so there's no mechanism to communicate the error back to the caller (without a breaking change)

Reply:
Ah. We'll leave it like this for now. We can revisit whether to move the error to `Build()` in the major (I think we probably should)